src/models/propulsion/FGBrushLessDCMotor.cpp

```diff
@@ -0,0 +1,236 @@
+  later version.
+
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+  details.
+
+  You should have received a copy of the GNU Lesser General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc., 59
+  Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+
+  Further information about the GNU Lesser General Public License can also be
+  found on the world wide web at http://www.gnu.org
+
+FUNCTIONAL DESCRIPTION
+--------------------------------------------------------------------------------
+Following code represent a new BrushLess DC motor to be used as alternative 
+to basic electric motor. 
+BLDC motor code is based on basic "3 constant motor equations"
+It require 3 basic physical motor properties:
+Kv speed motor constant
+Rm internal coil resistance
+I0 no load current
+
+HISTORY
+--------------------------------------------------------------------------------
+1/01/2022    Created
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+INCLUDES
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+#include <iostream>
+#include <sstream>
+#include <math.h>
+
+#include "FGFDMExec.h"
+#include "FGBrushLessDCMotor.h"
+#include "FGPropeller.h"
+#include "input_output/FGXMLElement.h"
+
+using namespace std;
+
+namespace JSBSim {
+
+/*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+CLASS IMPLEMENTATION
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
+
+FGBrushLessDCMotor::FGBrushLessDCMotor(FGFDMExec* exec, Element* el, int engine_number, struct FGEngine::Inputs& input)
+  : FGEngine(engine_number, input)
+{
+  Load(exec, el);
+
+  Type = etElectric;
+  PowerWatts = 745.7;
+  hptowatts = 745.7;
+
+
+  if (el->FindElement("maxvolts"))
+    MaxVolts = el->FindElementValueAsNumber("maxvolts");
+
+  if (el->FindElement("velocityconstant"))
+    VelocityConstant = el->FindElementValueAsNumber("velocityconstant");
+
+  if (el->FindElement("coilresistance"))
+    CoilResistance = el->FindElementValueAsNumber("coilresistance");
+  if (el->FindElement("noloadcurrent"))
+    NoLoadCurrent = el->FindElementValueAsNumber("noloadcurrent");
+  if (el->FindElement("deceleration_factor"))
+    DecelerationFactor = el->FindElementValueAsNumber("deceleration_factor");
+
+  MaxCurrent = MaxVolts / CoilResistance + NoLoadCurrent;
+
+  PowerWatts = MaxCurrent * MaxVolts;
+
+  string base_property_name = CreateIndexedPropertyName("propulsion/engine",
+    EngineNumber);
+  exec->GetPropertyManager()->Tie(base_property_name + "/power-hp", &HP);
+
+  exec->GetPropertyManager()->Tie(base_property_name + "/current-a", &CurrentRequired);
```

Comment:
```suggestion
  exec->GetPropertyManager()->Tie(base_property_name + "/current-a", &CurrentRequired);
```
According to your comment, wouldn't this property be better named `current-required_amperes` or something similar ?